### PR TITLE
Add doxygen into getdim.f Part of #55

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Doxygen has been added into getdim.f 
It is valided at
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/getdim_8f.html

### CODE REVIEW

File: src/getdim.f [hunk 1]
```diff
@@ -1,33 +1,24 @@
+!>    @file
+!>    @brief This routine converts grid information from a GRIB2 grid
```
Comment (GeorgeGayno-NOAA) [inline src/getdim.f]: @Hang-Lei-NOAA I don't see any references to GRIB1 in this routine. Is this description correct?

Comment (Hang-Lei-NOAA) [reply src/getdim.f]: This was not. updated

File: src/getdim.f [hunk 2]
```diff
@@ -1,33 +1,24 @@
+!>    @file
+!>    @brief This subroutine returns the info of GRIB2 Grid Definition
+!>    Section 3 format..
```
Comment (GeorgeGayno-NOAA) [inline src/getdim.f]: @Hang-Lei-NOAA I am getting this doxygen warning:  _warning: End of list marker found without any preceding list items_

If I remove the extra period on line 3, the warning is eliminated.

Comment (Hang-Lei-NOAA) [reply src/getdim.f]: Yes, it is updated. Thanks!

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Looks good. Approved.

**Pull request merged**